### PR TITLE
Remove bluez4 support, custom .pro build system and bump to 5.50. JB#57146

### PR DESCRIPTION
**Pull request by pvuorela**
Separate steps in commits:
- Get rid of bluez4 on the patches and separate .spec. All remaining code paths outside the "#if KF5BLUEZQT_BLUEZ_VERSION < 5" should be the same.
-  Get rid of custom .pro file build system and switch to cmake which the upstream is using. Required to pick pkg-config generation from newer version.
- Bump to 5.50 which was a step forward while not dealing yet with too big changes. After that there's some media interface stuff + small api adjustments that might be good to check more closely.

The pkg-config stuff affects depending packages, luckily not that many. Earlier was having code use #include <types.h> while with the current version that should be bluezqt/types.h. With such generic names the earlier was just bad.

@Tomin1 @spiiroin

### CODE REVIEW

**Comment by Tomin1** [comment]
README is a bit out of date now.

**Comment by pvuorela** [comment]
> README is a bit out of date now.

Oh yea, and there's a lefover .pro file there too. Maybe I'll just remove both.

File: kf5bluezqt.pro [hunk 1]
```diff
@@ -1,3 +0,0 @@
-TEMPLATE = subdirs
```
Comment (Tomin1) [inline kf5bluezqt.pro:1]: I think this removal was supposed to be part of "Build with cmake" commit. It doesn't matter much though.

**Review by Tomin1 — APPROVED** [review]
LGTM. One minor comment but approving already. :+1:

Soname versioning looks interesting but it's probably intended. :slightly_smiling_face:

File: kf5bluezqt.pro (hunk 1 above)
Comment (pvuorela) [reply kf5bluezqt.pro:1]: The actual usage was already removed with my earlier update PR so in a way it's not wrong in any commit :)

Comment (Tomin1) [reply kf5bluezqt.pro:1]: Ah, okay then. :slightly_smiling_face:

**Comment by pvuorela** [comment]
FWIW on my fedora the .so naming is similar, interpreting as intended.

**Pull request merged**